src/preamble.js

```diff
@@ -2223,6 +2223,7 @@ function integrateWasmJS(Module) {
       exports = instance.exports;
       if (exports.memory) mergeMemory(exports.memory);
       Module['asm'] = exports;
+      asm = exports;
```

Comment:
Because without it functions in js libraries that are marked `__asm` and use `setTempRet` (https://github.com/kripken/emscripten/blob/6dc4ac5f9e4d8484e273e4dcc554f809738cedd6/src/parseTools.js#L1296) look for it in the wrong place when we use async compilation, because we set global-level `asm` to the results of `doNativeWasm` which is empty, but we don't fix it up afterward.

That should probably be guarded by `#if WASM_BACKEND`.